### PR TITLE
wolfi-images: update mi2 base image

### PR DESCRIPTION
**Pull request by michaellzc**
## Test plan

tested in Cloud control plane dev environment.

everything worked as expected

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #56426** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56426" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56426?utm_source=stack-comment).

**Review by willdollman — APPROVED** [review]
lgtm! 

But! Cool new CI feature: if this PR contained the modifications to the `terraform.yaml` and `gke-gcloud-auth-plugin.yaml` packages then you don't need to add any of the the `@branch` stuff. 

The CI will:
* detect that you've modified the packages, build them and put them in the branch-specific repo
* ✨New✨ detect that this image contains those packages, build this image with the branch-specific repo added, and will hot-replace the `@sourcegraph` with `@branch` for you

Using stacked PRs so this won't work currently as the branch doesn't contain those changes, but once you merge the PRs in you can try it out. lmk if you hit any problems!

File: wolfi-images/cloud-mi2.yaml [hunk 1]
```diff
@@ -7,17 +7,20 @@ contents:
...
 
     ## mi2 packages
-    - terraform=1.3.9-r2
+    - terraform@sourcegraph
     - helm=3.11.3-r1
     - kustomize=4.5.7-r0
     - nodejs-18=18.17.1-r1
+    - kubectl-1.24=1.24.17-r1
+    - kubectl-1.24-default=1.24.17-r1
+    # TODO: change back to @sourcegraph
```
Comment (willdollman) [inline wolfi-images/cloud-mi2.yaml:16]:
```suggestion
```

File: wolfi-images/cloud-mi2.yaml [hunk 2]
```diff
@@ -7,17 +7,20 @@ contents:
...
+    - gke-gcloud-auth-plugin@sourcegraph
 
+    ## mi2 packages that we don't care about pinning version
+    - git
+    - git-lfs
+
+# this image is used on Cloud Run Job, and
+# it does not support running conatiner with non-root user.
```
Comment (willdollman) [inline wolfi-images/cloud-mi2.yaml:24]:
```suggestion
# it does not support running container with non-root user.
```

**Pull request merged**